packages/esm-patient-chart-app/src/clinical-views/encounter-tile/encounter-tile.component.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { useMemo } from 'react';
+import { useLayoutType } from '@openmrs/esm-framework';
+import { CodeSnippetSkeleton, Tile, Column, Layer } from '@carbon/react';
+import styles from './tile.scss';
+import { groupColumnsByEncounterType } from '../utils/helpers';
+import { useTranslation } from 'react-i18next';
+import { type EncounterTileColumn, type EncounterTileProps } from '../types';
+import { useLastEncounter } from '../hooks';
+
+export const EncounterTile = React.memo(({ patientUuid, columns, headerTitle }: EncounterTileProps) => {
+  const columnsByEncounterType = useMemo(() => groupColumnsByEncounterType(columns), [columns]);
+  const isTablet = useLayoutType() === 'tablet';
+
+  return (
+    <Layer className={styles.layer}>
+      <Tile className={styles.tile}>
+        <div className={isTablet ? styles.tabletHeading : styles.desktopHeading}>
+          <h4 className={styles.title}>{headerTitle}</h4>
+        </div>
+        <Column className={styles.columnContainer}>
+          {Object.entries(columnsByEncounterType).map(([encounterUuid, columns]) => (
+            <EncounterData
+              key={encounterUuid}
+              patientUuid={patientUuid}
+              encounterType={encounterUuid}
+              columns={columns}
+            />
+          ))}
+        </Column>
+      </Tile>
+    </Layer>
+  );
+});
+
+const EncounterData: React.FC<{
+  patientUuid: string;
+  encounterType: string;
+  columns: EncounterTileColumn[];
+}> = ({ patientUuid, encounterType, columns }) => {
+  const { lastEncounter, isLoading, error, isValidating } = useLastEncounter(patientUuid, encounterType);
+  const { t } = useTranslation();
+
+  if (isLoading || isValidating) {
+    return <CodeSnippetSkeleton type="multi" role="progressbar" />;
+  }
+
+  if (error || lastEncounter == undefined) {
+    return (
+      <div className={styles.tileBox}>
+        {columns.map((column, ind) => (
+          <div key={ind}>
+            <span className={styles.tileTitle}>{t(column.title)}</span>
+            <span className={styles.tileValue}>{error?.message}</span>
+          </div>
+        ))}
+      </div>
```

Comment:
If there's only a single error, why do we render the error once per column? I'd just display the error message in the box, maybe with something that says "Error loading most recent encounter" or something

Reply:
Each column can have a different encounter type, so an endpoint request will be made for each. One request can have an error and another successful, the purpose of displaying each per box is to help the person creating the columns to know why that specific value is not being displayed.